pkg/policy/repository.go

```diff
@@ -777,11 +787,23 @@ func (p *Repository) computePolicyEnforcementAndRules(securityIdentity *identity
 	}
 
 	matchingRules = []*rule{}
-	for _, r := range p.rules {
+	// Match cluster-wide rules
+	for rKey := range p.rulesByNamespace[""] {
+		r := p.rules[rKey]
 		if r.matchesSubject(securityIdentity) {
 			matchingRules = append(matchingRules, r)
 		}
 	}
+	// Match namespace-wide rules
+	namespace := lbls.Get(labels.LabelSourceK8sKeyPrefix + k8sConst.PodNamespaceLabel)
```

Comment:
What is the right behavior if a SecurityIdentity does not have a namespace?

Right now, we ensure that all namespaced Rules also have a subject selector for that namespace; it's what makes this possible. So, yes, this is safe given this assumption. The code that enforces this is [here](https://github.com/cilium/cilium/blob/main/pkg/k8s/apis/cilium.io/utils/utils.go#L336).

Thus, we should add a comment there that there is "far-away" code that relies on this.

Reply:
okay, I've added a comment above:
```
		// the MatchLabels map. Additionally, Policy repository relies
		// on this fact to properly choose correct network policies for
		// a given Security Identity.
```